src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -671,17 +629,161 @@ var ReactDOMFiberComponent = {
     }
 
     assertValidProps(tag, nextProps);
-    var wasCustomComponentTag = isCustomComponent(tag, lastProps);
-    var isCustomComponentTag = isCustomComponent(tag, nextProps);
+
+    var propKey;
+    var styleName;
+    var styleUpdates = null;
+    var updatePayload : null | Array<any> = null;
+    for (propKey in lastProps) {
+      if (nextProps.hasOwnProperty(propKey) ||
+         !lastProps.hasOwnProperty(propKey) ||
+         lastProps[propKey] == null) {
+        continue;
+      }
+      if (propKey === STYLE) {
+        var lastStyle = lastProps[propKey];
+        for (styleName in lastStyle) {
+          if (lastStyle.hasOwnProperty(styleName)) {
+            if (!styleUpdates) {
+              styleUpdates = {};
```

Comment:
Interestingly RN did this by packing all fields into one payload object. Caused a bunch of weird artifacts that made diffing harder to optimize.

We should reevaluate the actual algorithms here but for now I just wanted compatibility with CSSPropertyOperations since it is shared with Stack.